src/main/java/com/rcv/StreamingCVRReader.java

```diff
@@ -0,0 +1,340 @@
+  // param: columnAddress the input columnAddress String
+  // return: column index
+  public static int getColumnIndex(String columnAddress) {
+    // result is column index
+    int result = 0;
+    // i indexes over the "digits" of the columnAddress string
+    for (int i = 0; i < columnAddress.length(); i++) {
+      // at each iteration the current total will be multiplied by 26, "shifting" it left one place
+      result *= 26;
+      // charValue maps the current character to a value between 0 and 25
+      int charValue = columnAddress.charAt(i) - 'A';
+      assert(charValue >= 0 && charValue < 26);
+      result += charValue;
+    }
+    return result;
+  }
+
+  // function: handleEmptyCells
+  // purpose: Handle empty cells encountered while parsing a CVR.  Unlike empty rows, empty cells
+  // do not trigger parsing callbacks so their existence must be inferred and handled when they
+  // occur in a rankings cell.
+  // param: currentRank the rank at which we stop inferring empty cells.
+  void handleEmptyCells(int currentRank) {
+      // rank iterates between lastRankSeen and currentRank adding audit data and UWI rankings
+      for (int rank = lastRankSeen + 1;  rank < currentRank; rank++) {
+        // add data to audit log
+        currentCVRData.add("empty cell");
+        // add UWI ranking if required by settings
+        if (config.isTreatBlankAsUndeclaredWriteInEnabled()) {
+          Logger.log(Level.WARNING, "Empty cell -- treating as UWI");
+          // add the new ranking
+          currentRankings.add(new Pair<>(rank, config.getUndeclaredWriteInLabel()));
+        }
+      }
+  }
+
+  // function: beginCVR
+  // purpose: prepare to begin parsing a new CVR
+  void beginCVR() {
+    // setup data structures for parsing a new CVR
+    CVRIndex++;
+    currentRankings = new LinkedList<>();
+    currentCVRData = new LinkedList<>();
+    currentSuppliedCVRID = null;
+    currentPrecinct = null;
+    lastRankSeen = 0;
+  }
+
+  // function: endCVR
+  // purpose: complete construction of new CVR object
+  void endCVR() {
+    // handle any empty cells which may appear at the end of this row
+    handleEmptyCells(config.getMaxRankingsAllowed() + 1);
+    // determine what the new cvr ID will be
+    String computedCastVoteRecordID = String
+        .format("%s(%d)", StreamingCVRReader.this.excelFileName, CVRIndex);
+    // create new cast vote record
+    CastVoteRecord newRecord = new CastVoteRecord(
+        computedCastVoteRecordID, currentSuppliedCVRID, currentPrecinct,
+        currentCVRData, currentRankings);
+    // add it to overall list
+    CVRList.add(newRecord);
+    // provide some user feedback on the CVR count
+    if(CVRList.size() % 50000 == 0) {
+      Logger.log(Level.INFO, String.format("Parsed %d cast vote records ...", CVRList.size()));
+    }
+  }
+
+  // function: CVRCell
+  // purpose: handle a new CVR cell data
+  // param: col column of this cell
+  // param: row of this cell (unused)
```

Comment:
So why do you pass it?

Reply:
As I was building this code, it was the natural thing to do and I thought it might be useful for parsing other ballot styles.  But maybe it's time for row... _to go._